src/dataflow/source/kinesis.rs

```diff
@@ -46,11 +64,98 @@ where
         None
     };
 
-    // todo@jldlaughlin: Actually read from the Kinesis stream!
-    // Right now, this code will create a Kinesis source and immediately
-    // give up all its capabilities -- rendering it "Done".
-    let (stream, capability) = source(id, ts, scope, &name, move |_info| {
-        move |_cap, _output| SourceStatus::Done
+    let (stream, capability) = source(id, ts, scope, &name.clone(), move |info| {
+        let activator = scope.activator_for(&info.address[..]);
+
+        // Create a new AWS Kinesis Client.
+        let request_dispatcher = HttpClient::new().unwrap();
+        let provider = StaticProvider::new(access_key, secret_access_key, None, None);
+        let r: Region = region
+            .parse()
+            .unwrap_or_else(|_| panic!("Failed to parse AWS region: {}", region));
+        let client = KinesisClient::new_with(request_dispatcher, provider, r);
+
+        // Get the stream's name from the provided ARN.
+        let re = Regex::new(r"/(.*)").unwrap();
+        let captured_match = re.captures(&arn).unwrap().get(1); // do something better?
+        let stream_name = match captured_match {
+            Some(m) => m.as_str(),
+            None => panic!(
+                "Failed to read a stream name from the provided ARN: {}",
+                arn
+            ),
+        };
+
+        // todo@jldlaughlin: Read from multiple shards! #2222
+        let shards = block_on(get_shards_list(&client, stream_name));
```

Comment:
Similar to the `thread::sleep` comment below, we should chat about how blocking these block_on calls are. They may need to live on another thread.